src/Compilers/CSharp/Test/Syntax/Parsing/PatternParsingTests.cs

```diff
@@ -1900,9 +1900,15 @@ public void ParenthesizedExpression_03()
         public void ParenthesizedExpression_04()
         {
             UsingStatement(@"switch (e) { case (((x: 3))): ; }",
```

Comment:
Should we separate such tests into two? first to test if we emit the correct error message about the feature, and the second (which actually passes the feature flag) to assert actual diagnostics.
AFAIk, we shouldn't produce further diagnostics for unsupported expressions if that can be done.

Reply:
This is only intended to test parsing. It is not our practice to suppress further errors for features that are unsupported in the current version, as the follow-on errors are not necessarily cascaded diagnostics due to an error in the code.

---
In reply to: [168849349](https://github.com/dotnet/roslyn/pull/23357#discussion_r168849349) [](ancestors = 168849349)